src/http/one/TeChunkedParser.cc

```diff
@@ -125,6 +125,7 @@ Http::One::TeChunkedParser::parseChunkMetadataSuffix(Tokenizer &tok)
     // Code becomes much simpler when incremental parsing functions throw on
     // bad or insufficient input, like in the code below. TODO: Expand up.
     try {
+        tok.skipAll(CharacterSet::WSP); // Some servers send SP/TAB after chunk-size
```

Comment:
The proposed addition effectively disables BWS handling logic inside ParseBws() called from parseChunkExtensions() below. It also mishandles partial/incomplete input that ends exactly at chunk-size (Squid should not fall through to the next line in that case but does).

If we decide to allow these spaces, both bugs should be eliminated. Here is an untested suggestion:

```suggestion
        ParseBws(tok); // some servers send "chunk-size BWS CRLF"
```

The suggestion above also adjusts the comment to avoid implying that this code was added for all "SP/TAB after chunk-size" cases. Some of those cases are actually already covered by BWS handling in parseChunkExtension() call below.

Reply:
That was the original change I made locally, but with this change squid didn't even send the first chunk of data. 

Squid was failing on the 

```
        tok.skipRequired("CRLF after [chunk-ext]", Http1::CrLf());
```
in Http::One::TeChunkedParser::parseChunkMetadataSuffix(Tokenizer &tok);


```
2024/10/10 10:55:41.162 kid1| 11,2| http.cc(1466) decodeAndWriteReplyBody: de-chunking failure: cannot skip CRLF after [chunk-ext]
    exception location: Tokenizer.cc(159) skipRequired
```

Can't it be caused because `ParseBws(tok);` also removes CR char?

Reply:
Thank you for testing my broken suggestion even before I made it!

I now see why my suggestion was bad: When relaxed_header_parser is on, One::Parser::WhitespaceCharacters() includes CR. In this parsing context, skipping CR prevents subsequent `tok.skipRequired(...CrLf())` from succeeding. The existing chunk-ext parser code does not suffer from this problem because its BWS sequences are always followed by something (e.g., semicolon or extension name) _before_ we get to CRLF.

Unfortunately, this change request is still valid, even if the suggested simple solution is not. We will need to write more code to fix the problems this change request identifies. In summary, there will be two types of BWS -- one that relaxed_header_parser can extend and one that always contains just SP and HTAB.

Do you mind if I push a change to your PR branch that adds support for those two types of BWS?

Reply:
Yes, of course! Feel free to push it.

Yeah that's what I thought. It removed CR as well from the buffer and then it was not there when required for CrLf end of the chunk line.